odoo/tools/xml_utils.py

```diff
@@ -263,15 +263,25 @@ def validate_xml_from_attachment(env, xml_content, xsd_name, reload_files_functi
     """Try and validate the XML content with an XSD attachment.
     If the XSD attachment cannot be found in database, (re)load it.
 
+    A skip_xsd key can be provided in the context in order to skip the XSD validation.
+    This should be used during tests to avoid loading XSD files (and making http requests every time).
+
     :param odoo.api.Environment env: environment of calling module
     :param xml_content: the XML content to validate
     :param xsd_name: the XSD file name in database
+    :param reload_files_function: function that will be called to try and (re)load XSD files
     :return: the result of the function :func:`odoo.tools.xml_utils._check_with_xsd`
     """
+    if env.context.get('skip_xsd', False):
+        return
     try:
         _check_with_xsd(xml_content, xsd_name, env)
     except FileNotFoundError:
         if not reload_files_function:
             _logger.warning("You need to provide a function used to (re)load XSD files")
+            return
```

Comment:
So if there's no xsd file the entire thing is just silently ignored, with a mere warning in the logging?

What's wrong with letting the exception bubble up in that case (or swapping in a clearer exception)? If the caller wants to ignore the case of a missing xsd file, they can ignore the case of a missing xsd file.

Reply:
Yes, the goal here is to avoid blocking the user for a file not found, which could simply occur if the internet connection is unstable or unavailable (= not downloading XSD files).
Another scenario is when the server from which we want to download the XSD files is down, or the files are no longer available (outdated url for instance, which means a newer XSD is most likely available). In such a case, I feel like it's better to let it go for now - at worst the generated XML is wrong and the file will be refused, but it has nothing to do with the XSD since the implementation of the XML is most likely outdated as well.